multimodal/vl2l/src/mlperf_inference_multimodal_vl2l/evaluation.py

```diff
@@ -0,0 +1,176 @@
+        true_categories = [c.strip() for c in true_path.split(separator)]
+
+        # Check for empty paths
+        if not predicted_categories or not true_categories:
+            return 0, len(predicted_categories), len(true_categories)
+
+        # 2. Count the intersection (longest common prefix)
+        intersection_count = 0
+
+        # Iterate through the paths simultaneously
+        for pred_cat, true_cat in zip(predicted_categories,
+                                      true_categories,
+                                      strict=False):
+            if pred_cat == true_cat:
+                intersection_count += 1
+            else:
+                # Stop as soon as a mismatch is found (enforces hierarchical match)
+                break
+
+        pred_length = len(predicted_categories)
+        true_length = len(true_categories)
+
+        return intersection_count, pred_length, true_length
+
+    def calculate_hierarchical_metrics(self, data: list) -> float:
+        """Calculates the aggregate hP, hR, and hF scores for a list of samples.
+
+        Args:
+            data: A list of tuples, where each tuple is
+                (predicted_path_str, true_path_str).
+
+        Returns:
+            F1 score
+        """
+        total_intersection = 0
+        total_predicted_length = 0
+        total_true_length = 0
+
+        # 1. Aggregate the components across all samples
+        for pred_path, true_path in data:
+            intersection, pred_len, true_len = \
+                self.get_hierarchical_components(pred_path, true_path)
+
+            total_intersection += intersection
+            total_predicted_length += pred_len
+            total_true_length += true_len
+
+        # 2. Calculate hP and hR
+        hp = total_intersection / total_predicted_length \
+            if total_predicted_length > 0 else 0.0
+        hr = total_intersection / total_true_length \
+            if total_true_length > 0 else 0.0
+
+        return 0.0 if hp + hr == 0 else 2 * (hp * hr) / (hp + hr)
+
+    def calculate_exact_match(self, generated_text: str, original_text: str) -> float:
+        """Calculates binary Exact Match (EM) score.
+
+        We clean the text (lowercase, strip whitespace) for a fairer comparison.
+
+        Args:
+            generated_text: Output from the VLM.
+            original_text: Ground truth information from the dataset.
+
+        Returns:
+            1 if the values match or 0 otherwise
+        """
+        gen = generated_text.strip().lower()
+        orig = original_text.strip().lower()
+
+        return 1.0 if gen == orig else 0.0
+
+
+    def run_evaluation(self) -> None:
+        """Main function to run the evaluation."""
+        with Path.open(self.filename) as f:
+            model_output = json.load(f)
+
+        original_data = load_dataset(
+            self.dataset.repo_id,
+            token=self.dataset.token,
+        )["train"]
```

Comment:
Related to [this comment](https://github.com/mlcommons/inference/pull/2393/files#r2543688271), we should parametrize which split of the dataset this is for.